run.py

```diff
@@ -33,17 +37,80 @@ def _get_opt_parser(self):
         parser.add_option('-p', '--pass', dest='password', default=None)
         return parser
 
-    def _get_connection(self):
+    def _get_connection(self, hostname, username, password):
```

Comment:
It might be possible to gather this information from juju.  The juju ssh unit already knows the credentials.  I just don't know how we can get them